ActionMenu/ActionMenu.cs

```diff
@@ -689,6 +700,29 @@ public static Menus SplitOverCrowdedMenus(Menus m, uint overcrowdedMenuValue = 7
             return item;
         }
 
+        private static readonly System.Text.RegularExpressions.Regex impulseRe = new System.Text.RegularExpressions.Regex(
+            @"\s*<impulse(?:=(?<duration>\d+(?:\.?\d*)))>$",
+            System.Text.RegularExpressions.RegexOptions.IgnoreCase | System.Text.RegularExpressions.RegexOptions.Compiled
+            );
+
+        private static (string control, float? duration) DetectImpulseToggle(CVRAdvancedSettingsEntry s)
+        {
+            if (s.machineName.EndsWith("Impulse")) return (control: "impulse", duration: null);
+            var match = impulseRe.Match(s.name);

```

Comment:
This should be s.machineName . If I remember s.name is the display name, but we want to match the name used in animator.